### DISCUSSION TITLE
How can I validate the validate the form depending on if the specific values are chosen from Select dropdown in React hook form?

### DISCUSSION THREAD

**Discussion by lakhandeshpande** [react-hook-form/react-hook-form#3220] [General]
I have a select field as follows

`       <Select
                     type="select"
                      label={label}
                      ref={ !checkvalidation  ? register({ required: 'Please select at least one workday option' }) : register({ required: false }) }
                       watch={watch}
                       name={name}
                       size="sm"
                       defaultValue=""
                       onChange={validateWorkPattern(watch, employmentId)}
     >`

On value change I am checking if specific value selected else should be return false for setting required to true.

But the validateWorkPattern function is taking long and slowing down the performance.

How can I achieve this without impacting performance. 

I would also like to know if I could do it onSubmit but how to execute the onsubmit?

**Comment by bluebill1049** [comment]
`validate` + `getValues()`

**Reply by lakhandeshpande** [reply]
Could you help me with demo this thing? I am being novice to react hook forms

**Reply by bluebill1049** [reply]
```
register({ validate: () => { 
  return getValues().test === 'bill'
}}
```

**Reply by lakhandeshpande** [reply]
Let me have a try with this one.

**Reply by bluebill1049** [reply]
and make sure you read the doc :) they r all there.

**Reply by lakhandeshpande** [reply]
I am doing this : 
`ref={ register({ validate: () => { 
                                            const fname = day.fieldname
                                             return getValues()[employmentId].fname === 'NONE' ?  { required: true } : {required: false}
                                          }})}
`

But it still allows me to proceed even if I select NONE

**Reply by bluebill1049** [reply]
read register: https://react-hook-form.com/api#register there is example for validate there
